### ISSUE TITLE
Inconsistency between bernoulli and bernoulli_

### ISSUE THREAD

**Issue by hengyuan-hu** [pytorch/pytorch#1259]
```
import torch                                                                               
                                                                                           
a = torch.FloatTensor(3, 5).uniform_(0, 1)                                                 
a.round_()                                                                                 
                                                                                           
print a.bernoulli()                                                                        
print a.bernoulli_()                                                                       
```
Tensor a is rounded to 0 or 1 so the bernoulli sampling should be deterministic. However, two printed bernoulli are different.

**Comment by albanD** [comment]
I am not sure this is the intended behaviour but this is expected.
The full function signature is `torch.bernoulli(input, out=None) → Tensor`
When you do `a.bernoulli()`, this is equivalent to `torch.bernoulli(a)`.
When you do `a.bernoulli_(a)`, this is equivalent to `torch.bernoulli(a, out=a)` (what you want in your case I think)
When you do `a.bernoulli_(float_val=0.5)`, this function fills each entry of `a` is a bernouilli with probability `float_val` which has a default value of 0.5.

To answer your question, when you do `a.bernoulli_()`, this is equivalent to `a.bernoulli_(0.5)` and not `a.bernoulli_(a)` as you were expecting.

**Comment by hengyuan-hu** [comment]
Great! Thank you for your clarification. The `a.bernoulli_(float_val=0.5)` semantic is somewhat misleading as I was expecting bernoulli_ takes itself as the parameter just like bernoulli but just do a in-place replacement instead of create a new tensor. Also the default arg `float_val=0.5` is not mentioned in the doc, which is a bit misleading.

**Comment by Smerity** [comment]
I just ran into this as well, including the lack of documentation clarification with regards to how `bernoulli_` actually operates. Before raising an issue / asking in the Slack channel, I went hunting as it seemed an obvious oversight and found this issue. If I get the chance I'll look into filling in the docs more.

**Comment by 3rd3** [comment]
Maybe off-topic, but where can I find the full signatures of such functions? The docs only seem to mention that one can broadcast one scalar probability to all tensor entries.

**Comment by janchorowski** [comment]
I just got bitten by this inconsistency as well. Maybe the doc has to be updated to make it more clear? Even giving an example that `a = torch.bernoulli(a)` is the in-place equivalent of `a.bernoulli_(a)` but not `a.bernoulli_()` which is equivalent to `a.bernoulli_(0.5)`?

**Comment by zou3519** [comment]
This definitely needs better documentation

**Comment by martinbrose** [comment]
I think this might have been closed with PR #10273?

Current documentation states the following, which looks like it is addressing the above:
<img width="1083" alt="Screen Shot 2020-08-06 at 4 52 17 pm" src="https://user-images.githubusercontent.com/13284268/89553362-532b3c80-d805-11ea-8017-f32caad5d6cf.png">

**Comment by zou3519** [comment]
@martinbrose thanks for pointing that out. We indeed have good bernoulli docs now so I am closing this issue.
